crypto/x509/x_pubkey.c

```diff
@@ -309,14 +363,30 @@ int X509_PUBKEY_set(X509_PUBKEY **x, EVP_PKEY *pkey)
  */
 static int x509_pubkey_decode(EVP_PKEY **ppkey, const X509_PUBKEY *key)
 {
-    EVP_PKEY *pkey = EVP_PKEY_new();
+    EVP_PKEY *pkey;
+    int nid;
+
+    nid = OBJ_obj2nid(key->algor->algorithm);
+    if (!key->flag_force_legacy) {
```

Comment:
Should that `!` be there?

Reply:
Yes. This function only does legacy loading, so unless we're forcing legacy loading we only want to continue if we have an ENGINE for this key type.